internal/reconcilers/status.go

```diff
@@ -0,0 +1,235 @@
+	if err != nil && apierrors.IsConflict(err) {
+		log.Info("conflict updating status, requeuing")
+		return true, nil
+	}
+	if err != nil {
+		log.Error(err, "error updating status")
+		return true, err
+	}
+
+	log.Info("updated status")
+	return false, nil
+}
+
+func (r *StatusReconciler) setStatusInfo(status v1.ConditionStatus, reason string, message string, condition v1.Condition) v1.Condition {
+	condition.Status = status
+	condition.Reason = reason
+	condition.Message = message
+	return condition
+}
+
+func (r *StatusReconciler) conditionStatus(b bool) v1.ConditionStatus {
+	if b {
+		return v1.ConditionTrue
+	}
+	return v1.ConditionFalse
+}
+
+func (r *StatusReconciler) reconcileImageStreamStatus(ctx context.Context, pipeline *certv1alpha1.OperatorPipeline, indexType, indexName string) (bool, error) {
+	readyCondition := v1.Condition{
+		Type:               fmt.Sprintf("%sReady", indexType),
+		ObservedGeneration: pipeline.Generation,
+		Status:             v1.ConditionUnknown,
+	}
+	log := r.Log.WithValues("status.observedGeneration", pipeline.Generation)
+
+	imageStream := &imagev1.ImageStream{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: pipeline.Namespace, Name: indexName}, imageStream)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.WithValues("imagestream", types.NamespacedName{Namespace: pipeline.Namespace, Name: indexName}).
+			Error(err, "failed to get object")
+		return true, err
+	}
+
+	if err != nil && apierrors.IsNotFound(err) {
+		meta.SetStatusCondition(&pipeline.Status.Conditions, r.setStatusInfo(
+			r.conditionStatus(false),
+			"NotFound",
+			fmt.Sprintf("%s with name %s not found", indexType, indexName),
+			readyCondition))
+		err = r.Client.Status().Update(ctx, pipeline, &client.UpdateOptions{})
+		if err != nil && apierrors.IsConflict(err) {
+			log.Info("conflict updating object, requeueing")
+			return true, nil
+		}
+		if err != nil {
+			log.Error(err, "failed to update object")
+			return true, err
+		}
+	}
+
+	meta.SetStatusCondition(&pipeline.Status.Conditions, r.setStatusInfo(
+		r.conditionStatus(true),
+		"AsExpected",
+		fmt.Sprintf("%s with name %s found", indexType, indexName),
+		readyCondition))
+
+	return false, nil
+}
+
+func (r *StatusReconciler) reconcileSecretStatus(ctx context.Context, pipeline *certv1alpha1.OperatorPipeline, secretType, secretName, secretKey string) (bool, error) {
+	// Kubeconfig Secret
+	readyCondition := v1.Condition{
+		Type:               fmt.Sprintf("%sReady", secretType),
+		ObservedGeneration: pipeline.Generation,
+		Status:             v1.ConditionUnknown,
+	}
+	log := r.Log.WithValues("status.observedGeneration", pipeline.Generation)
+	secret := &corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: pipeline.Namespace, Name: secretName}, secret)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.WithValues("kubeconfigsecret", types.NamespacedName{Namespace: pipeline.Namespace, Name: secretName}).
```

Comment:
Shouldn't this be `secreteType` or `secreteName` here?

Reply:
Good catch.